### DISCUSSION TITLE
Full charge interval

### DISCUSSION THREAD

**Discussion by z-master42** [reinhard-brandstaedter/solarflow-control#106] [Q&A] [answered]
Based on the config.ini example, I understand that solarflow-control tries to get the battery to 100 % for 32 hours. This means that there is no feed-in from the SolarFlow during this time. After the battery is full or after 32 hours, feed-in is based on demand.

When does the interval start again, i.e. when will solarflow-control try to get the battery full again within 32 hours?

**Comment by reinhard-brandstaedter** [comment, marked as answer]
Hi @z-master42 ,

no that is not how it works. Solarflow-control will try to ensure the battery gets charged every now and then. Since you cannot predict how fast the battery gets fully charged there is no fixed interval for charging but a target maximum interval. (32 hours).
If the battery is not fully charged within 32hours (the last full charge is longer than 32 hours ago). No discharging will happen to ensure the battery gets charged as fast as possible.
Assuming that the battery gets usually fully charged before the 32h limit (or whatever you set, depending on your setup), you will get a "use stored energy as quick as possible"-behavior.

This is especially targeted for setups with panels connected to SF and panels connected "directly" to the inverter. During low-light periods (fall/winter) this ensures the battery really gets charged to 100% (maintaining cell drift and battery health), and at the same time ensures that SF is used to support the house demand when needed during nigth/dawn/dusk times in general good seasons (summer).

Additionally to the above there are two parameters that control a guaranteed charging power (min_charge_power). SF will only contribute to household if more solar power is generated than this min_charge_power (given the above battery full condition is met)
The second is max_discharge_power. This is the maximum power that SF will contribute to the household if needed. This is also the power level at which SF will feed in during night, when there is no input from direct connected panels. "Night" is determined by current sunrise/sunset times +/- the sunrise/sunset_offset parameters.

Here is an example of how this looks in reality:
Note that this setup uses currently a very low power (365Wp) panel connected to SF, so the charging time to get from 0 to 100% are indeed very long (additionally to the bad weather).
blue bars below zero is the charging power
blue lines above zero is the power input to home by direct panels
red bars above zero is the discharging of SF and feed into home
<img width="1583" alt="image" src="https://github.com/reinhard-brandstaedter/solarflow-control/assets/10830223/5d643f94-5667-4ed1-b3cd-acc5852bd6ff">


I hope that makes things clearer.

**Reply by z-master42** [reply]
Thank you very much for this detailed explanation. I think I have now understood the intention to some extent. Let's see how it works for me. There really isn't much yield at the moment.
